### PR TITLE
Use Slack incoming-webhooks to post messages

### LINKED ISSUE #212: Use Slack incoming-webhooks to post messages
**Description**

As we're switching to the Camunda Slack, the way it is setup, IT doesn't allow apps to use the channels:write permission but instead requires us to use an [incoming webhook](https://slack.dev/java-slack-sdk/guides/incoming-webhooks). 

I've already created it, the new app, the new token, etc., but I think we need to slightly modify the API usage.

### PR DESCRIPTION
**Pull request by npepinpe**
**Description**

This PR removes the real-time-messaging Slack integration and replaces it with a simple incoming-webhook usage. See the original issue as to why.

The self test is replaced with a simple request sending an empty payload which should return a specific error. This is somewhat brittle, and welcome to be challenged (not sufficient to check the error code as using a weird URL will still return a 400, just a different error).

Vault has been updated with a new `slackWebhookUrl`, and the docs have also been updated (partially - it seems other parts also need to be updated but were outside of this scope).

**Related issues**

closes #212

### CODE REVIEW

**Comment by npepinpe** [comment]
I've deployed a test workflow so I could test notifications. In dev you can start it as:

```sh
zbctl create instance Process_0h3atq5 --variables "{\"testWorkflowId\": \"the test workflow ID\", \"clusterPlan\": \"the cluster plan\", \"generation\": \"the generation\", \"clusterName\": \"the cluster name\", \"operateURL\": \"https://237f9938-e974-4f5d-9494-d42a7d8fac1f.operate.dev.ultrawombat.com\", \"testReport\": {\"failureCount\": 1, \"failureMessages\": [\"failure message 1\", \"failure message 2\"]}}"
```

And you should see your message in #zeebe-slack-dev. That seemed the fastest way to test this, let me know if there's another way I should've tested it (it does pollute the dev cluster with an additional workflow I guess).

**Review by pihme — APPROVED** [review]
One tiny change, then it's good to go for deployment.

Congratulations on your first PR for the testbench :tada:

File: docs/operator-handbook.md [hunk 1]
```diff
@@ -80,18 +80,18 @@ Setup and deployment:
...
-1. Download token to authenticate the app
-1. Give it permissions to write to channels
+1. Create an incoming webhook for your app (https://api.slack.com/messaging/webhooks)
+1. Add a new webhook to your workspace for the desired channel (you may need approval from your workspace admin)
 1. Invite your app to the channel it should publish to
 1. Fill deployment descriptors as follows:
 
 | File           | Field              | Content                         |
 | -------------- | ------------------ | ------------------------------- |
 | testbench.yaml | ZCTB_SLACK_CHANNEL | The slack channel to publish to |
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/operator-handbook.md]: Oops, I had twice merge conflicts when rebasing and I messed up once and had to manually correct, so I'm not surprised something still made it through :see_no_evil: Thanks for reading carefully!

**Pull request merged**